v1/ao/internal/reporter/reporter_test.go

```diff
@@ -548,39 +389,57 @@ func TestInvalidKey(t *testing.T) {
 
 }
 
-func TestRetryDelay(t *testing.T) {
-	c := &grpcConnection{}
-	retries := grpcMaxRetries + 1
-	newDelay, err := c.setRetryDelay(1, &retries)
-	assert.Equal(t, ErrMaxRetriesExceeded, err)
-	assert.Equal(t, 0, newDelay)
-
-	retries = 0
-	oldDelay := 500
-	newDelay, err = c.setRetryDelay(oldDelay, &retries)
-	expectedDelay := int(500 * grpcRetryDelayMultiplier)
-	assert.Equal(t, expectedDelay, newDelay)
-	assert.Equal(t, 1, retries)
-
-	retries = 0
-	oldDelay = 50000
-	newDelay, err = c.setRetryDelay(oldDelay, &retries)
-	expectedDelay = grpcRetryDelayMax * 1000
-	assert.Equal(t, expectedDelay, newDelay)
-	assert.Equal(t, 1, retries)
+func TestDefaultBackoff(t *testing.T) {
+	var backoff []int64
+	expected := []int64{
+		500, 750, 1125, 1687, 2531, 3796, 5695, 8542, 12814, 19221, 28832,
+		43248, 60000, 60000, 60000, 60000, 60000, 60000, 60000, 60000}
+	bf := func(d time.Duration) { backoff = append(backoff, d.Nanoseconds()/1e6) }
+	for i := 1; i <= grpcMaxRetries+1; i++ {
+		DefaultBackoff(i, bf)
+	}
+	assert.Equal(t, expected, backoff)
+	assert.NotNil(t, DefaultBackoff(grpcMaxRetries+1, func(d time.Duration) {}))
+}
+
+type NoopDialer struct{}
+
+func (d *NoopDialer) Dial(c grpcConnection) (*grpc.ClientConn, error) {
+	return nil, nil
 }
 
 func TestInvokeRPC(t *testing.T) {
```

Comment:
The `TestInvokeRPC` function is performing multiple tests(reporter exiting, invalid service key, no retry, invocation error, etc) on grpc call. In case of any failure, identifying what test has failed is not easy. Breaking this function into multiple tests might be beneficial when identifying a failure and to reduce the complexity of this test function.